cls/pkg/isc/codetidy/CodeTidy.cls

```diff
@@ -129,556 +153,1219 @@ ClassMethod ProcessStream(ByRef instream, lang = "COS", pInfo)
+						"JAVASCRIPT":next("Type") = $$$JAVASCRIPTWhiteSpace, 
+						"SQL":next("Type") = $$$SQLWhiteSpace, 
+						:"") {
+						do ..ReadFragment(.last, .this, .next)
+						set currentWhiteSpace = currentWhiteSpace _ this("Frag")
+					}
+					
+					if next("Lang") '= "\n" {
+						// Insert new line and opening whitespace
+						set this("Lang") = "\n"
+						set this("Type") = ""
+						set this("Frag") = ""
+						do ..InsertFragment(.last, .this)
+						do ..ParseLine(.last, .this, .next, .conv)
+						
+						set this("Lang") = this("Lang")
+						set this("Type") = $case(this("Lang"), 
+							"COS":$$$COSWhiteSpace, 
+							"CSS":$$$CSSWhiteSpace, 
+							"HTML":$$$HTMLWhiteSpace, 
+							"JAVASCRIPT":$$$JAVASCRIPTWhiteSpace, 
+							"SQL":$$$SQLWhiteSpace, 
+							:"")
+						set this("Frag") = ##class(pkg.isc.codetidy.Utils).GetIndentString()
+						do ..InsertFragment(.last, .this)
+						do ..ParseFragment(.last, .this, .next, .conv)
+					}
+				}
+			}
+		}
+		quit
+	}
+	// &javascript
+	if this("Type") = $$$COSJavascript {
+		set conv("Frag") = this("Frag")
+		if $zconvert(conv("Frag"), "L") = "js" set conv("Frag") = "javascript"
+		if ##class(pkg.isc.codetidy.Utils).GetUseCapitals() = 0 {
+			set conv("Frag") = $zconvert(conv("Frag"), "l")
+		} else {
+			set conv("Frag") = $zconvert(conv("Frag"), "w")
+		}
+		do ..ApplyIndentation(.this, .next, .conv, "<", ">")
+		
+		set conv("Line") = conv("Line") _ conv("Frag")
+		quit
+	}
+	// &html
+	if this("Type") = $$$COSHTML {
+		set conv("Frag") = $zconvert(this("Frag"), "L")
+		set conv("Line") = conv("Line") _ conv("Frag")
+		quit
+	}
+	if this("Type") = $$$COSPreProcessorCommand {
+		set conv("Frag") = $zconvert(this("Frag"), "L")
+		set conv("Line") = conv("Line") _ conv("Frag")
+		if ##class(pkg.isc.codetidy.Utils).GetIndent() {
+			if $case(this("Frag"), "if":1, "ifdef":1, "ifndef":1, :0)  {
+				set next("Indent") = next("Indent") + 1
+			} elseif $case(this("Frag"), "else":1, "elseif":1, :0) {
+				set this("Indent") = this("Indent") - 1
+				set next("Indent") = next("Indent") + 1
+			} elseif $case(this("Frag"), "endif":1, :0) {
+				set this("Indent") = this("Indent") - 1
+			}
+		}
+		quit
+	}
+	if this("Type") = $$$COSComment {
+		set conv("Frag") = this("Frag")
+		do ..ParseFragmentCOSComment(.last, .this, .next, .conv)
+		
+		set conv("Line") = conv("Line") _ conv("Frag")
+		quit
+	}
+	
+	// Function or Command or Special Variables
+	if (this("Type") = $$$COSFunction) || (this("Type") = $$$COSCommand) || (this("Type") = $$$COSSystemvariable) {
+		set conv("Frag") = this("Frag")
+		
+		if (this("Type") = $$$COSCommand) {
+			set conv("PostConditional") = 0
+		}
+		
+		set conv("Frag") = $get($$$MappedValue(this("Lang"), this("Type"), $zconvert(conv("Frag"), "L")), conv("Frag"))
+		
+		if ##class(pkg.isc.codetidy.Utils).GetUseCapitals() = 0 {
+			set conv("Frag") = $zconvert(conv("Frag"), "l")
+		} else {
+			set conv("Frag") = $zconvert(conv("Frag"), "w")
```

Comment:
This seems like we're overriding the mapped value with "word" case

Reply:
Really the solution here is to rely on $$$MappedValue rather than doing additional case conversion

Reply:
Although we could do something like check:
```
if $Data($$$MappedValue(this("Lang"), this("Type"), $zconvert(conv("Frag"), "L")), caseCorrected)#2 {
    set conv("Frag") = caseCorrected
} elseif ##class(pkg.isc.codetidy.Utils).GetUseCapitals() = 0 {
    set conv("Frag") = $zconvert(conv("Frag"), "l")
} else {
    set conv("Frag") = $zconvert(conv("Frag"), "w")
}
```